### DISCUSSION TITLE
Asset Modules, publicPath and filename: double slash in request url

### DISCUSSION THREAD

**Discussion by rikers** [webpack/webpack#17715] [Q&A]
I'm facing a weird issue regarding my prod config when using "Asset modules" in my webpack.prod.js. 
While inspecting request URLS in the browser, everything works as expected. The main bundle, chunks and assets are in all 200.
But I noticed that the request URL for my fonts and images (see the 2 rules using  type: 'asset/resource') have a double slash in the url. 

**Here are two examples:**

... dist//images ...
http://my-awesome-theme.local/wp-content/themes/my-awesome-theme/dist//images/shadow-top.7134fc9e352e11618552.png

... dist//fonts ...
http://my-awesome-theme.local/wp-content/themes/my-awesome-theme/dist//fonts/awesomet-webfont.19100ce8b46300853317.woff2

I assumed that adding an empty string for the public path would fix the issue, but no.

Anybody encountered this issue ? thank you in advance!

**packages versions:**

"webpack": "^5.88.2"
"webpack-cli": "^5.1.4"
"webpack-manifest-plugin": "^5.0.0"

**Configuration file**

`module.exports = (env, argv) => merge(common(env, argv), {
    mode: 'production',
    devtool: UNCOMPRESSED ? 'source-map' : false,

    output: {
        path: path.resolve(__dirname, 'dist'),
        publicPath: 'wp-content/themes/my-awesome-theme/dist/',
        filename: 'js/scripts-[name].[contenthash].js',
        chunkFilename: 'js/chunks/[name].[contenthash].js',
        clean: true
    },

    optimization: {
        minimize: UNCOMPRESSED ? false : true,
    },

    module: {
        rules: [
            {
                test: /\.(js)$/,
                include: [path.resolve(__dirname, 'src', 'js')],
                use: ['babel-loader'],
            },
            {
                test: /\.s[ac]ss$/i,
                include: [path.resolve(__dirname, 'src', 'scss')],
                use: [MiniCssExtractPlugin.loader, 'css-loader', 'sass-loader'],
            },
            // @See: https://webpack.js.org/guides/asset-modules/
            // emits a separate file and exports the URL. Previously achievable by using file-loader
            {
                test: /\.(png|svg|jpg|jpeg|svg|gif)$/i,
                include: [path.resolve(__dirname, 'src', 'images')],
                type: 'asset/resource',
                generator: {
                    publicPath: '',
                    filename: 'images/[name].[hash][ext]'
                },
            },
            {
                test: /\.(woff|woff2|ttf|eot)$/i,
                include: [path.resolve(__dirname, 'src', 'fonts')],
                type: 'asset/resource',
                generator: {
                    publicPath: '',
                    filename: 'fonts/[name].[hash][ext]'
                },
            },
        ],
    },

    plugins: [
        new MiniCssExtractPlugin({filename: 'css/styles-[name].[contenthash].css'}),
        new WebpackManifestPlugin()
    ],
});
`

**Comment by ekr3peeK** [comment]
Noticing the same issue, with similar config. I do not know what is adding the extra slash, the browser I am developing in can handle it correctly, but I am a bit affraid that this will not be the case for all browsers on the market.

**Comment by xak2000** [comment]
Also found this behavior and in my case the resulting url with double-slash is actually incorrect (the server returns 500).

Strangely, adding double slash to the start of the `publicPath` string solves the issue.
E.g.:

```js
      {
        test: /\.(png|jpe?g|gif|svg)$/i,
        type: 'asset/resource',
        generator: {
          outputPath: 'images/',
          // For some reason without leading `//` the resulting path in .css files will contain an extra slash.
          // E.g.: background-image: url(..//images/1952df896c6a79bb730c.png?1);
          publicPath: '//images/'
        }
      }
```

**Comment by alexander-akait** [comment]
If you still have a problem, please create reproducible test repo, sorry for delay
